### PR TITLE
Allow binding UNIX domain sockets

### PR DESCRIPTION
**Pull request by spladug**
I really don't love `bind.rb`, so if you have any thoughts on how to do that better I'm all ears.

There's also some duplication in the "clean up an existing socket" stuff between this code and the command socket stuff. I wasn't sure how to avoid that while maintaining the descriptive error messages.

:eyeglasses: @ajacksified

### CODE REVIEW

File: lib/einhorn/bind.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
+require 'fcntl'
```
Comment (kjoconnor) [inline lib/einhorn/bind.rb]: Is this used here?

Comment (spladug) [inline lib/einhorn/bind.rb]: nope. removed.

File: bin/einhorn [hunk 2]
```diff
@@ -181,6 +181,20 @@ EOF
...
+  # we accept two types of socket address
+  (?:
+    # ip, host:port
+    (?:(?<host>[^:]+):(?<port>\d+)) |
+    # unix socket path
+    (?<path>[^,:]+)
+  )
+
+  # flags are optional, comma delimited, and come at the end
+  (?<flags>(?:,\w+)*)
```
Comment (rram) [inline bin/einhorn:195]: :no_bell: this regex just makes me feel uncomfortable. I feel like it should start with a slash, or be restricted to less characters.

Comment (spladug) [inline bin/einhorn:195]: did you mean to comment on the unix socket path part?

Comment (rram) [inline bin/einhorn:195]: derp. yea

Comment (spladug) [inline bin/einhorn:195]: hm. i was gonna add a starting slash requirement, but then i was thinking that maybe someone would want to do relative paths?

Comment (rram) [inline bin/einhorn:195]: We know they would be wrong anyway. :stuck_out_tongue: 
Right now, what if they wanted to make a path with a comma or colon in it?! Spaces and unicode would work. I really don't care that much. Give em enough rope to hang themselves with!

Comment (spladug) [inline bin/einhorn:195]: I think I'll let the regex be agnostic and `bind` can take care of validating it's a meaningful path.

**Comment by spladug** [comment]
gonna try this against upstream.

**Pull request closed without merging**